### PR TITLE
PREL-166 add proxy support

### PR DESCRIPTION
**Pull request by vorsel**

### CODE REVIEW

File: scripts/percona-release.sh [hunk 1]
```diff
@@ -20,6 +20,21 @@ if [[ -f /etc/default/percona-release ]]; then
     source /etc/default/percona-release
 fi
 
+# Special proxy handling for cURL
+CURL_PROXY=
+CURL_EXEC=( curl )
```
Comment (EvgeniyPatlan) [inline scripts/percona-release.sh:25]: what avout using absolute path?

File: scripts/percona-release.sh [hunk 2]
```diff
@@ -293,15 +308,25 @@ function create_yum_repo {
...
 function create_apt_repo {
   local _repo=${1}
+  local _proxyline=
+
   [[ ${1} = "original" ]] && _repo=percona
   REPOURL="${URL}/${_repo}/apt ${CODENAME}"
+  if [[ -n "${CURL_PROXY}" ]]; then
+      _proxyline="Acquire::${APT_PROXY_SCHEME}::Proxy::${URL/${APT_PROXY_SCHEME}:\/\//} \"${CURL_PROXY}\";"
+
+      if [[ ! -f /etc/apt/apt.conf.d/99-percona-release ]] || ! grep -Rq "${_proxyline}" /etc/; then
```
Comment (EvgeniyPatlan) [inline scripts/percona-release.sh:326]: brackets are missed

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

File: scripts/percona-release.sh (hunk 1 above)
Comment (vorsel) [reply scripts/percona-release.sh:25]: done

File: scripts/percona-release.sh (hunk 2 above)
Comment (vorsel) [reply scripts/percona-release.sh:326]: done

**Review by AgileStas — APPROVED** [review]
These changes should work, so approving them.

However, I think that original issue was created with wrong assumptions. Please see https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/ and curl commit https://github.com/curl/curl/blob/30e7641d7d2eb46c0b67c0c495a0ea7e52333ee2/lib/url.c#L2250-L2261
So, the only change that is needed, is to set http_proxy to HTTP_PROXY when the first one is unset and the second one is set.
Also, I'm not sure if setting permanent proxy for the repository based on env variable is a good idea. Think about location-dependent proxy: one for workplace, another one during traveling, etc.)

**Review by adivinho — APPROVED** [review]

**Review by EvgeniyPatlan — APPROVED** [review]

**Pull request merged**